plasmapy/physics/transport.py

```diff
@@ -205,3 +209,1493 @@ def Coulomb_logarithm(T, n_e, particles, V=None):
+            'Spitzer-Harm',     described in Ref [2].
+            'Epperlein-Haines', described in Ref [3]. <not yet implemented>
+            'Ji-Held',          described in Ref [4].
+
+    field_orientation : string
+        Either of 'parallel', 'par', 'perpendicular', 'perp', 'cross', or
+        'all', indicating the cardinal orientation of the magnetic field with
+        respect to the transport direction of interest. Note that 'perp' refers
+        to transport perpendicular to the field direction in the direction of
+        the temperature gradient, while 'cross' refers to the B X grad(T)
+        direction. The option 'all' will return a numpy array of all three,
+        np.array((par, perp, cross)).
+
+    coulomb_log_ei: float or dimensionless Quantity, optional
+        Force a particular value to be used for the Coulomb logarithm. If
+        None, the PlasmaPy function Coulomb_Logarithm() will be used. Useful
+        for comparing calculations.
+
+    V_ei: Quantity, optional
+       Supplied to coulomb_logarithm() function, not otherwise used.
+       The relative velocity between particles.  If not provided,
+       thermal velocity is assumed: :math:`\mu V^2 \sim 3 k_B T`
+       where `mu` is the reduced mass.
+
+    coulomb_log_ii: float or dimensionless Quantity, optional
+        Force a particular value to be used for the Coulomb logarithm. If
+        None, the PlasmaPy function Coulomb_Logarithm() will be used. Useful
+        for comparing calculations.
+
+    V_ii: Quantity, optional
+       Supplied to coulomb_logarithm() function, not otherwise used.
+       The relative velocity between particles.  If not provided,
+       thermal velocity is assumed: :math:`\mu V^2 \sim 3 k_B T`
+       where `mu` is the reduced mass.
+
+    hall_e: float or dimensionless Quantity, optional
+        Force a particular value to be used for the e Hall parameter. If
+        None, the PlasmaPy function Hall_parameter() will be used. Useful
+        for comparing calculations.
+
+    hall_i: float or dimensionless Quantity, optional
+        Force a particular value to be used for the ion Hall parameter. If
+        None, the PlasmaPy function Hall_parameter() will be used. Useful
+        for comparing calculations.
+
+    mu: optional, float or dimensionless Quantity
+        mu = m_e / m_i
+        Ji-Held model only, may be used to include ion-electron effects
+        on the ion transport coefficients. Defaults to m_e / m_i.
+        Set to zero to disable these effects.
+
+    theta: optional, float or dimensionless Quantity
+        theta = T_e / T_i
+        Ji-Held model only, may be used to include ion-electron effects
+        on the ion transport coefficients. Defaults to T_e / T_i. Only
+        has effect if mu is non-zero.
+
+    Raises
+    ------
+    Um, I'm not sure at the moment. ValueError?
+
+    PhysicsError
+
+    Returns
+    -------
+    transport_coeffs: list of Quantity
```

Comment:
Since this is probably (am I right?) not going to be a long calculation to get each of the parameters here, could we perhaps calculate all of them by default and return a `namedtuple` instead? Perhaps it could work like this:
    
    transport = classical_transport(...)
    print(transport.resistivity)
    print(transport.thermoelectric_conductivity)

and so on.

I could implement this if you think this is reasonable.

If we think the calculation is going to be computationally expensive, another approach would be to have `classical_transport` return an object of a `transport` class that could then calculate the transport coefficients on demand, same syntax as before, except that we'd use a `@property` to calculate it on demand.

Reply:
Hmm. Well, for right now, it shouldn’t be expensive. In future, if used for a simulation, it may be called a lot for each grid and time steps. One might want to not incorporate, say, viscosity, so I thought it’d be nice to start with the idea that you might not want all of them. Hence the list thing. I’ll be the first to admit it’s not elegant, so Im open to better solutions, but I do like the idea of not having to calculate them all. Eventually we probably need to handle numpy arrays.

Reply:
Sure, that's a good point about having to handle numpy arrays. I guess I'll try the second option, with the transport class and properties?

Reply:
Hmm, could the object-orientedness somehow be optional?  I don't like objects. :)

Reply:
They're handy sometimes :D I can get a test implementation working that shows the concept but doesn't use much object oriented features.

Reply:
Sounds good to me!